src/proofs/ProofSteps.cpp

```diff
@@ -178,17 +178,14 @@ void StepHandler::buildAletheProof() {
     for (std::size_t i = 0; i < derivationSize; ++i) {
         auto const & step = derivation[i];
 
-        // We don't need the simplification steps of the previous derivation procedure
-        simplificationSteps.clear();
-
         if (step.premises.empty()) { continue; }
 
         std::vector<int> requiredMP;
 
         if (!modusPonensSteps.empty()) {
             // Get the necessary steps for modus ponens
-            for (std::size_t i = 0; i < step.premises.size(); i++) {
-                requiredMP.push_back(modusPonensSteps[(int)step.premises[i] - 1]);
+            for (unsigned long premise : step.premises) {
+                requiredMP.push_back(modusPonensSteps[(int)premise - 1]);
```

Comment:
Do not use C-style casts.

Looking at this code, I think in several places, we should be using `std::size_t` instead of `int`.
For example, `modusPonensSteps` should be `std::vector<std::size_t>>` not vector of `ints`.
Similarly, `currStep` should be `std::size_t`, it can never be negative.
While at it, please rename `currStep` to `currentStep`.

Reply:
Also, just use `auto premise` here, instead of `unsigned long premise`.